### ISSUE TITLE
Needs a better way to select the CRS

### ISSUE THREAD

**Issue by AlisterH** [linz/linz-data-importer#47]
If I want to load e.g. the Auckland Lidar DEM WMS (not WMTS), there is apparently a vast number of CRSs available (perhaps this is an error though, because they seem too many).
They are all in a list with just the EPSG numbers (not ideal, but people can probably cope with it), but the list isn't sorted, so it is impossible to find the one you want - see screenshot.

**Comment by AlisterH** [comment]
![image](https://user-images.githubusercontent.com/714600/51653504-8b3ea600-1ff8-11e9-9c84-1e3a59d24da7.png)

**Comment by AlisterH** [comment]
> there is apparently a vast number of CRSs available (perhaps this is an error though, because they seem too many).

I think there are fewer if you try in QGIS 2.18.x rather than 3.x, but it still seems quite a lot.

**Comment by SPlanzer** [comment]
Thanks @AlisterH 

The WMS capabilities document lists all these as "All supported EPSG projections".
With WMTS and WFS only one or two are ever available. 

The purpose of the EPSG combo-box is to show the user which crs's the data is served by the server (if they are in the list they are stored in this project and do not need to be reprojected - This is useful for the user to understand as the user can then understand when QGIS will reporject on the fly. (this statement is true for WMTS and WFS though considering the number for WMS I will need to investigate this for further for WMS)) . 

Currently the plugin:
* Ensures the projects current crs is selected in the combo-box. I expect In most cases the user will not alter this (?)

Will do to improve the plugin:
* Ensure the list is ordered

Consider:
* how to reduce the number
* Consider adding a 'search and filter' text box in the combo-box

**Comment by AlisterH** [comment]
> Currently the plugin: Ensures the projects current crs is selected in the combo-box


Ah, I see it does do this in QGIS 2, but it doesn't in QGIS 3.

**Comment by SPlanzer** [comment]
Custom combobox widget built with text filtering of items.

However, when promoting the custom combobox widget, the plugin fails at start up. 

The below test results in the same error when Service_dialog_base.ui is extended by the custom combo box error. related to https://issues.qgis.org/issues/8760?
```
import sys, os
from PyQt5 import QtCore, QtGui, uic


FORM_CLASS, _ = uic.loadUiType(os.path.join(
    os.path.dirname(__file__), "/home/splanzer/git/linz-data-importer/linz-data-importer/gui/Service_dialog_base.ui"))
class MyWindowClass(QtGui.QMainWindow, form_class):
    def __init__(self, parent=None):
        QtGui.QMainWindow.__init__(self, parent)
        self.setupUi(self)
app = QtGui.QApplication(sys.argv)
myWindow = MyWindowClass(None)
myWindow.show()
app.exec_()
```

**Comment by SPlanzer** [comment]
The above was related to bad import paths

**Comment by SPlanzer** [comment]
using python sort() on the wms wspg list returned from the server each time is to slow
```
with ordering
2019-03-12T11:03:25     INFO    loaded: data.linz.govt.nz wmts in 0.013724803924560547secs
2019-03-12T11:03:28     INFO    loaded: data.linz.govt.nz wms in 2.6306211948394775secs
2019-03-12T11:03:29     INFO    loaded: data.linz.govt.nz wfs in 0.014017581939697266secs
2019-03-12T11:03:29     INFO    loaded: data.mfe.govt.nz wmts in 0.002277374267578125secs
2019-03-12T11:03:29     INFO    loaded: data.mfe.govt.nz wfs in 0.004220724105834961secs
2019-03-12T11:03:31     INFO    loaded: data.linz.govt.nz wmts in 0.0039033889770507812secs

without ordering
2019-03-12T11:04:32     INFO    loaded: data.linz.govt.nz wmts in 0.009961366653442383secs
2019-03-12T11:04:32     INFO    loaded: data.linz.govt.nz wms in 0.0009458065032958984secs
2019-03-12T11:04:33     INFO    loaded: data.linz.govt.nz wfs in 0.0087432861328125secs
2019-03-12T11:04:33     INFO    loaded: data.mfe.govt.nz wmts in 0.0012078285217285156secs
2019-03-12T11:04:33     INFO    loaded: data.mfe.govt.nz wfs in 0.0025501251220703125secs
2019-03-12T11:04:35     INFO    loaded: data.linz.govt.nz wmts in 0.002384185791015625secs
```

The server has a predefined espg list for all wms layers. Therefore it will be ordered once and applied as the crs espg option for all layers.

**Comment by SPlanzer** [comment]
Applying order to WMS ESPG list for each domain is obviously much faster

```
2019-03-12T11:48:08     INFO    loaded: data.linz.govt.nz wmts in 0.01619410514831543secs
2019-03-12T11:48:09     INFO    loaded: data.linz.govt.nz wms in 0.007515907287597656secs
2019-03-12T11:48:10     INFO    loaded: data.linz.govt.nz wfs in 0.017155885696411133secs
2019-03-12T11:48:10     INFO    loaded: data.mfe.govt.nz wmts in 0.003507852554321289secs
2019-03-12T11:48:10     INFO    loaded: data.mfe.govt.nz wfs in 0.00481867790222168secs
```

**Comment by SPlanzer** [comment]
resolved via pull request #52 

Still to be released

**Comment by SPlanzer** [comment]
New version (2.0.1) released with enhancement
